### PR TITLE
resolved: optimize dbus change notification signals a bit

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
Split out of #17535

**Comment by yuwata** [comment]
> Prompted-by: #17577 (but doesn't fix this, since this commit only
handles D-Bus-induced changes, not the ones made via networkd)

Fixes in networkd side are #18320 and #18422.

File: src/basic/set.h [hunk 1]
```diff
@@ -152,3 +152,5 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(Set*, set_free_free);
 #define _cleanup_set_free_free_ _cleanup_(set_free_freep)
 
 int set_strjoin(Set *s, const char *separator, bool wrap_with_separator, char **ret);
+
+bool set_equal(Set *a, Set *b);
```
Comment (bluca) [inline src/basic/set.h:156]: could you please add a simple unit test for this in src/test/test-set.c ?

Comment (poettering) [reply src/basic/set.h:156]: done

**Comment by poettering** [comment]
Force pushed a new version. Only change: added an extensive test, as suggested. Upgrading green label

**Comment by poettering** [comment]
CI failure appears unrelated.

**Pull request merged**